app/src/main/java/ch/sdp/vibester/activity/TypingGameActivity.kt

```diff
@@ -246,9 +247,28 @@ class TypingGameActivity : AppCompatActivity() {
         else{
             if(runnable !=null ){
                 h.removeCallbacks(runnable!!);}
-            val i = Intent(this, EndBasicGameTemporary::class.java)
-            i.putExtra("score", gameManager.getScore().toString())
-            startActivity(i)
+            switchToEnding(gameManager)
         }
     }
+
+    private fun switchToEnding(gameManager:GameManager) {
+        val intent = Intent(this, GameEndingActivity::class.java)
+        val incArray: ArrayList<String> = ArrayList(gameManager.getWrongSongs().map{it.getTrackName() +" - "+ it.getArtistName()})
+
+        val statNames: ArrayList<String> = arrayListOf()
+        val statName = "Total Score"
+        statNames.addAll(arrayOf(statName,statName,statName,statName,statName))
+
+        val statVal: ArrayList<String> = arrayListOf()
+        val score = gameManager.getScore().toString()
+        statVal.addAll(arrayOf(score, score, score, score, score))
+
+        intent.putExtra("nbIncorrectSong", gameManager.gameSize - gameManager.getScore())
+
+        intent.putStringArrayListExtra("str_arr_inc", incArray)
+        intent.putStringArrayListExtra("str_arr_name", statNames)
+        intent.putStringArrayListExtra("str_arr_val", statVal)
+
+        startActivity(intent)
+    }
```

Comment:
Maybe refactor the creations of the ArrayListand the calls to putStringArrayListExtra into a separate function? (Unless the values we will use instead of the hardcoded ones are of types other than String). The rest of the function looks good to me.

Reply:
True, maybe I will mode the creation of the arraylist with the delimeter '-' inside the gameManager.... I will taake a note for myself for the next pr. thanks!